### PR TITLE
Start arch doc

### PR DESCRIPTION
**Pull request by daved**
https://www.pivotaltracker.com/story/show/179459065

### CODE REVIEW

File: architecture.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+### locale/
+
+Localization keys and associated values.
+
+### pkg/
+
+Packages that are made available for use by any application, including external
+code. The packages that are currently located here are not intended for use by
+external code, and will eventually be moved into the `internal/` directory.
```
Comment (Naatan) [inline architecture.md:94]:
```suggestion
code. Note we are changing our strategy with regards to these packages, 
and these will eventually be moved into the `internal/` directory.
```

File: architecture.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+#### pkg/cmdlets/
+
+Packages that are made available for use by "runner" packages. Synonymous with
+`internal/runbits/`, and all new runner-common packages should be placed there.
```
Comment (Naatan) [inline architecture.md:99]: Other way around; runbits is new, cmdlets are old.

File: architecture.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+#### pkg/platform/
+
+Packages focused on interacting with the platform. Much of the behavior is
```
Comment (Naatan) [inline architecture.md:103]:
```suggestion
Packages focused on interacting with platform API's. Much of the behavior is
```

File: architecture.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+generated, but there are also critical components providing platform-dependent
+client-side logic (e.g. `pkg/platform/runtime`).
+
+#### pkg/{project,projectfile}/
+
+Packages that provide setup and interaction with the activestate.yaml files.
+
+### scripts/
+
+Helper scripts.
```
Comment (Naatan) [inline architecture.md:113]:
```suggestion
Helper scripts for development and deployment processes.
```

**Review by Naatan — CHANGES_REQUESTED** [review]

File: architecture.md (hunk 2 above)
Comment (daved) [reply architecture.md:99]: That is what I tried to convey, that runbits is the location of all new common packages. Should I word it differently? Maybe:
```suggestion
#### pkg/cmdlets/

Packages that are made available for use by "runner" packages. Synonymous with
`internal/runbits/` with all new runner-common packages being placed there.
```

Comment (Naatan) [reply architecture.md:99]: I think you should make the title `pkg/runbits` and instead hint at cmdlets. runbits are the way forward, so it should take center stage.

Comment (daved) [reply architecture.md:99]: That directory is covered: https://github.com/ActiveState/cli/pull/1556/files#diff-4e03c430fc7f575c72fd989625acdc7d8d70fd20d3693c893d67d9dbd461ccdbR77-R81

Comment (daved) [reply architecture.md]: Just pushed a change to remove the pkg/cmdlets entry and update the runbits entry.

**Review by Naatan — APPROVED** [review]

**Review by Naatan — APPROVED** [review]

**Pull request merged**